knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/impl/TemplateEngine.java

```diff
@@ -53,15 +56,23 @@ private void initHandlebars() {
         return extractFragments(request.getTemplate())
                 .doOnNext(this::traceSnippet)
                 .flatMap(this::compileHtmlFragment)
-                .concatMapEager(item ->
-                        snippetProcessor.processSnippet(item, request)
+                .filter(htmlFragment -> shouldProcessRequest(htmlFragment, request))
```

Comment:
I think filteringOut will cause that you will miss filtered out snippet in resulting String, is it desired ?